### PR TITLE
docs(keyconcepts): Documenting changes to notification policy with automatic integration

### LINKED ISSUE #397: Document the changes and undoing those changes for the provider configurations
**Is your feature request related to a problem? Please describe.**
Some of the providers will try to automagically set up themselves for webhook/push mode. This could give an ease of use for some, but I was a bit afraid about not knowing what would happen to my accounts, if I grant a high permission to Keep and let it modify stuff. 
It also introduced an issue that, if I have my environment for example for Datadog or Cloudwatch in IaC (terraform for example), it might make changes that are not in my source control and it will get rolled back on the next run. Or, if I have 500 alarms outside of IaC and all get a small modifications from the providers, I will need to manually undo things, if I want to delete the provider for some reason.

**Describe the solution you'd like**
It would be nice to document what will be modified exactly and to have the delete provider option either recommend undo steps, or do them if the setup was automatic.

### PR DESCRIPTION
**Pull request by mobley-trent**
Closes #397
/claim #397

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

**Comment by mobley-trent** [comment]
Hi @talboren I'm trying to understand the scope of the issue here. The automatic webhook integration adds Keep to certain scopes within the provider. Is this the change meant to be documented ? The documentation on the Keep UI seems to show this as a sidenote.

Screenshot 2024-06-07 180359

**Review by talboren — APPROVED** [review]
LGTM!

**Comment by Matvey-Kuk** [comment]
@talboren I think the issue is about "document what will be modified exactly" meaning exact changes for each integration 🤔

**Comment by talboren** [comment]
> @talboren I think the issue is about "document what will be modified exactly" meaning exact changes for each integration 🤔

Sorry, just noticed it's a part of this issue and it's a draft.

**Comment by talboren** [comment]
> Hi @talboren I'm trying to understand the scope of the issue here. The automatic webhook integration adds Keep to certain scopes within the provider. Is this the change meant to be documented ? The documentation on the Keep UI seems to show this as a sidenote.
> 
> Screenshot 2024-06-07 180359

@mobley-trent please see @Matvey-Kuk's comment. I think the greater issue here is to document exactly what each provider is changing within the provider (for instance, in Grafana, we add a new contact point and notification policy)

File: docs/providers/documentation/grafana-provider.mdx [hunk 1]
```diff
@@ -69,3 +69,9 @@ You can check that the Grafana Provider works by testing Keep's contact point (w
   <img height="10" src="/images/grafana_sa_5.png" />
 </Frame>
 5. Go to Keep - you should see an alert from Grafana!
+
+## Webhook Integration Modifications
+
+The webhook integration adds Keep as a contact point in the Grafana instance. Keep also gains access to the following scopes:
+- `alert.provisioning:read`
+- `alert.provisioning:write`
```
Comment (mobley-trent) [inline docs/providers/documentation/grafana-provider.mdx:77]: @talboren is this the correct format ? If so, I shall proceed to add this side note to every provider that supports the webhook integration.

Comment (Matvey-Kuk) [reply docs/providers/documentation/grafana-provider.mdx:77]: Yep! It's worth mentioning how to find this contact point and what its name is. By the way, I see there are more scopes here https://docs.keephq.dev/providers/documentation/grafana-provider#connecting-with-the-provider. Do we ask for unnecessary scopes by mistake?

Comment (talboren) [reply docs/providers/documentation/grafana-provider.mdx:77]: > Yep! It's worth mentioning how to find this contact point and what its name is. By the way, I see there are more scopes here https://docs.keephq.dev/providers/documentation/grafana-provider#connecting-with-the-provider. Do we ask for unnecessary scopes by mistake?

I think it's just the screenshot I took before clicking on Apply ^^'

**Review by Matvey-Kuk — APPROVED** [review]

**Comment by mobley-trent** [comment]
@talboren I'd appreciate your feedback

**Comment by talboren** [comment]
@mobley-trent @Matvey-Kuk this original issue also mentions “how to undo” the changes, do you think it’s within the scope of this PR? If not, this LGTM and @Matvey-Kuk can hit the merge button!

**Comment by mobley-trent** [comment]
Hi @Matvey-Kuk I was just wondering about the bounty 😅

**Pull request merged**